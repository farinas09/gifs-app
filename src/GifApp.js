import React, {useState} from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifApp = ({defaultCategories = []}) => {
    const [categories, setCategories] = useState(defaultCategories)

    return (
        <div>
            <h2>Gif App</h2>
            <hr/>
            <AddCategory setCategories={setCategories} />
            <ol>
                {
                    categories.map( (category, i) => <GifGrid key={category} category={category}/>) 
                }
            </ol>
        </div>
    )
}
