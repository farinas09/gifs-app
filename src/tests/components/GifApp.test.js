import React from "react"

import '@testing-library/jest-dom';
import { shallow } from 'enzyme'
import { GifApp } from "../../GifApp";

describe('Pruebas en <GifApp />', () =>{

    test('should show the app component successfully', () => {
        
        
    const wrapper = shallow(<GifApp />)
        expect( wrapper ).toMatchSnapshot();
    })

    test('should show a list of categories', () => {
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GifApp defaultCategories = {categories} />)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
        
    })
    
})