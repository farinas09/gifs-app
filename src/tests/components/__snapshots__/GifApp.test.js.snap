// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifApp /> should show a list of categories 1`] = `
<div>
  <h2>
    Gif App
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</div>
`;

exports[`Pruebas en <GifApp /> should show the app component successfully 1`] = `
<div>
  <h2>
    Gif App
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</div>
`;
